### PR TITLE
implement the interface ArrayableInterface for BaseActiveRecord

### PR DESCRIPTION
**Pull request by niqingyang**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ❌
| Fixed issues  | https://github.com/yiisoft/active-record/issues/272#issue-2021848888 # fixed by the PR

### CODE REVIEW

**Comment by terabytesoftw** [comment]
The tests fails.

File: src/BaseActiveRecord.php [hunk 1]
```diff
@@ -1266,19 +1270,28 @@ public function getTableName(): string
         return $this->tableName;
     }
 
-    public function toArray(): array
+    /**
+     * @inheritDoc
+     */
+    public function toArray(array $fields = [], array $expand = [], bool $recursive = true): array
```
Comment (Tigrov) [inline src/BaseActiveRecord.php:1276]:
```suggestion
```
Looks like implementation from `ArrayableTrait` also good.

**Review by Tigrov — CHANGES_REQUESTED** [review]
`ActiveRecordInterface::fields()` also should be removed as a part of `ArrayableInterface`

https://github.com/niqingyang/active-record/blob/fa7c4cb1144c5d2190ddeb37d4c773bc4921312b/src/ActiveRecordInterface.php#L89-L95

File: tests/Driver/Oracle/ActiveRecordTest.php [hunk 2]
```diff
@@ -166,4 +167,58 @@ public function testToArrayWithClosure(): void
...
+                'status',
+                'item.id',
+                'item.name',
+                'item.email',
+                'items.0.id',
+                'items.0.name',
+                'items.0.email',
+            ]),
+        );
+    }
```
Comment (Tigrov) [inline tests/Driver/Oracle/ActiveRecordTest.php:223]:
```suggestion
```

`\Yiisoft\ActiveRecord\Tests\Driver\Oracle\ActiveRecordTest` inherits from `\Yiisoft\ActiveRecord\Tests\ActiveRecordTest` where the test already exists.

File: tests/Driver/Pgsql/ActiveRecordTest.php [hunk 3]
```diff
@@ -381,4 +382,58 @@ public function testToArrayWithClosure(): void
...
+                'status',
+                'item.id',
+                'item.email',
+                'item.name',
+                'items.0.id',
+                'items.0.email',
+                'items.0.name',
+            ]),
+        );
+    }
```
Comment (Tigrov) [inline tests/Driver/Pgsql/ActiveRecordTest.php:438]:
```suggestion
```

`\Yiisoft\ActiveRecord\Tests\Driver\Pgsql\ActiveRecordTest` inherits from `\Yiisoft\ActiveRecord\Tests\ActiveRecordTest` where the test already exists.

**Review by Tigrov — CHANGES_REQUESTED** [review]

File: src/ActiveRecordInterface.php [hunk 4]
```diff
@@ -4,15 +4,15 @@
...
 use Throwable;
+use Yiisoft\Arrays\ArrayableInterface;
 use Yiisoft\Db\Exception\Exception;
 use Yiisoft\Db\Exception\InvalidArgumentException;
 use Yiisoft\Db\Exception\InvalidConfigException;
 use Yiisoft\Db\Exception\NotSupportedException;
 use Yiisoft\Db\Exception\StaleObjectException;
 
-interface ActiveRecordInterface
+interface ActiveRecordInterface extends ArrayableInterface
```
Comment (Tigrov) [inline src/ActiveRecordInterface.php:15]:
```suggestion
interface ActiveRecordInterface
```

It is better to move ArrayableInterface to implementation of BaseActiveRecord as you did before

File: src/BaseActiveRecord.php [hunk 5]
```diff
@@ -42,6 +43,7 @@
  */
 abstract class BaseActiveRecord implements ActiveRecordInterface, IteratorAggregate, ArrayAccess
```
Comment (Tigrov) [inline src/BaseActiveRecord.php:44]:
```suggestion
abstract class BaseActiveRecord implements ActiveRecordInterface, IteratorAggregate, ArrayAccess, ArrayableInterface
```

**Review by Tigrov — CHANGES_REQUESTED** [review]
👍

File: src/ActiveRecordInterface.php (hunk 4 above)
Comment (niqingyang) [reply src/ActiveRecordInterface.php]: ok

File: src/BaseActiveRecord.php (hunk 5 above)
Comment (niqingyang) [reply src/BaseActiveRecord.php]: ok

File: src/ActiveRecordInterface.php [hunk 6]
```diff
@@ -502,6 +502,16 @@ public function populateRecord(array|object $row): void;
...
+     * in {@see fields()} will be returned.
+     * @param array $expand the additional fields that the output array should contain.
+     * Fields not specified in {@see extraFields()} will be ignored. If this parameter is empty, no extra fields
+     * will be returned.
+     * @param bool $recursive Whether to recursively return array representation of embedded objects.
+     *
+     * @return array The array representation of the object.
      */
-    public function toArray(): array;
+    public function toArray(array $fields = [], array $expand = [], bool $recursive = true): array;
```
Comment (Tigrov) [inline src/ActiveRecordInterface.php:516]: As it is a part of `ArrayableInterface`, the methods `fields()` and `toArray()` no needed in `ActiveRecordInterface`

Comment (niqingyang) [reply src/ActiveRecordInterface.php]: ok

**Review by Tigrov — APPROVED** [review]

**Comment by Tigrov** [comment]
Thank you 👍

**Comment by niqingyang** [comment]
> Thank you 👍

🤝

**Pull request merged**